packages/framework-observable/README.md

```diff
@@ -0,0 +1,152 @@
+# @neos-project/framework-observable
```

Comment:
thank you for providing extensive documentation for these framework parts.

I think we already partially discussed the idea of the "The Client-side framework" from https://github.com/neos/neos-ui/pull/3331:

> ### `@neos-project/framework-observable`
> 
> The `@neos-project/framework-observable` package provides a very simple implementation of the `Observable`-pattern (see: https://github.com/tc39/proposal-observable).
> 
> Furthermore, based on `Observable`s, it establishes two key primitives for use in the UI:
> 
> #### State
> 
> A `State` observable represents a value that changes over time. It can be asked about its current value and any subscriber will receive the current value immediately.
> 
> It also provides an `update` method, that allows to push the next value calculated from the current value.
> 
> #### Channel
> 
> A `Channel` observable represents a sort of message bus. A subscriber will receive all messages that have been published to the `Channel` since the subscription was established.
> 
> It provides a `publish` method to push messages onto the channel.
> 
> ### `@neos-project/framework-react-hooks`
> 
> The `@neos-project/framework-react-hooks` package contains a bunch of essential React hooks to allow binding state to React components.
> 
> Most notably it contains the `useLatestValueFrom` hook, that allows to bind `Observable`s to components.

And for the DTO serialisation and de-serialisation (`@neos-project/framework-schema`) you elaborated why it makes sense to provide a custom implementation instead of using a npm library: https://github.com/neos/neos-ui/pull/3331#issuecomment-1518032438

Regarding observers and state handling i remember having talked to you regarding using rx-js vs a custom implementation like `@neos-project/framework-observable` and i think the reasons for a custom implementation were customisability (limited feature scope we only need to implement what we need), smaller bundle size, and that  `Observable` will hopefully come to native ECMA script which we definitely want to profit from if available and this can be ensured by having it all under control.
But im a little fuzzy on the details and dont quite remember it all. So it would be great to discuss this to fill in the gaps. In the end this adds complexity to our codebase and thus the benefits and downsides should be carefully considered. When we decide for this library an outlined discussion would also help future maintainers to understand our decision ;) 

EDIT:
I just skimmed over the implementation and its a super slim layer so +1 from my side ;) And we could consider this dicussion closed ^^ The documentation and tests are more than the code :D

Reply:
You've summarized it perfectly :)